api/v1/Layer.php

```diff
@@ -772,6 +773,34 @@ public function latestRaster(float $time): PromiseInterface
         );
     }
 
+    /**
+     * @throws Exception
+     */
+    public function getExportLayerDescriptions(
+        string $workspace,
+        string $layer
+    ): array {
+        $response = $this->geoserver->ows(
+            $workspace
+            . "/ows?service=WMS&version=1.1.1&request=DescribeLayer&layers="
+            . urlencode($workspace)
+            . ":"
+            . urlencode($layer)
+            . "&outputFormat=application/json"
+        );
```

Comment:
I saw ows() can also return null on error.

```suggestion
        if (null === $response = $this->geoserver->ows(
            $workspace
            . "/ows?service=WMS&version=1.1.1&request=DescribeLayer&layers="
            . urlencode($workspace)
            . ":"
            . urlencode($layer)
            . "&outputFormat=application/json"
        )) {
            $data = [];
            $data["layerDescriptions"] = [];
            $data["error"] = 'Failed to export layer descriptions. No response.';
            return $data;
        }
```

Another way to do it, is to let ows() just throw exceptions. It is currently catching error and printing it... and returning null:
```
        try {
            Log::LogInfo("Calling geoserver: ".$this->baseurl.$url);
            $result = $this->CallBack(
                $this->baseurl . $url,
                array(),                // no content to send
                array(),                // no headers to send
                false,                  // sync request, so wait for it
                false,                  // no content, so no json encoding of it required either
                $customOpt
            ); // see above additional curl opts for this request
        } catch (Throwable $e) {
            print("Geoserver request failed to url ".$url.". Exception: ".$e->getMessage().PHP_EOL);
            return null;
        }
```

I would remove the try catch in ows() and just let the exception be catched later. 
Then you can try catch it on the ows() calls themselves, eg:
(with the benefit that you show the reason of the error)
```
    try {
        $response = $this->geoserver->ows(
            $workspace
            . "/ows?service=WMS&version=1.1.1&request=DescribeLayer&layers="
            . urlencode($workspace)
            . ":"
            . urlencode($layer)
            . "&outputFormat=application/json"
        );    
    } catch (Throwable $e) {
          $data = [];
          $data["layerDescriptions"] = [];
          $data["error"] = 'Failed to export layer descriptions. No response. Reason: ' . $e->getMessage();
          return $data;       
    }
```

Well of course the latter is more work ... see what you can do.

Reply:
Yes, totally, I removed the try-catch block around the ows function to let it throw exceptions.

Reply:
Say ows will throw an exception now. If I understand correctly it will be catched by Router.php on cally the GetExport method of the Layer class in : 
```
            try {
                // ....
                $payload = $class->$method(...$arguments);
                // ....
            } catch (Throwable $e) {
                $success = false;
                // ....
                $message = Base::ErrorString($e);
                return self::formatResponse($success, $message, null, $className, $method, $data);
            }
```

So that would return an array from formatResponse with:
```
            "header_type" => null,
            "header_data" => null,
            "success" => false,
            "message" => $message, //<--- the exception message
            "payload" => null
```

So this is correct it seems.